src/BotPlutusInterface/CardanoNode/Query.hs

```diff
@@ -0,0 +1,78 @@
+-- | Several query functions to query local node
+module BotPlutusInterface.CardanoNode.Query (
+  NodeQueryError (..),
+  NodeConn,
+  QueryConstraint,
+  queryInCardanoMode,
+  queryBabbageEra,
+  toQueryError,
+  connectionInfo,
+) where
+
+import BotPlutusInterface.Types (PABConfig (..))
+import Cardano.Api qualified as CApi
+import Control.Monad.Freer (Eff, LastMember, Member, send)
+import Control.Monad.Freer.Reader (Reader, ask)
+import Control.Monad.Trans.Class
+import Control.Monad.Trans.Either
+import Control.Monad.Trans.Except (throwE)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import System.Environment (getEnv)
+import Prelude
+
+{- | Error returned in case any error happened querying local node
+ (wraps whatever received in `Text`)
+-}
+data NodeQueryError
+  = NodeQueryError Text
+  deriving stock (Eq, Show)
+
+type NodeConn = CApi.LocalNodeConnectInfo CApi.CardanoMode
+
+type QueryConstraint effs = (Member (Reader NodeConn) effs, LastMember IO effs)
+
+queryInCardanoMode ::
```

Comment:
This is for me. Add the docs to explain clearly what these functions do.

Reply:
fixed by https://github.com/mlabs-haskell/bot-plutus-interface/pull/103/commits/bd88edc3d37d48352abedf82a826a8be6cf91797